app/src/main/java/com/mk/tasky/authentication/registration/presentation/RegistrationViewModel.kt

```diff
@@ -1,19 +1,79 @@
 package com.mk.tasky.authentication.registration.presentation
 
+import android.util.Log
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.mk.tasky.authentication.domain.AuthenticationRepository
+import com.mk.tasky.authentication.registration.domain.RegistrationUseCase
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class RegistrationViewModel @Inject constructor(
-    private val repository: AuthenticationRepository
+    private val repository: AuthenticationRepository,
+    private val registrationUseCase: RegistrationUseCase
 ) : ViewModel() {
-    init {
+    var state by mutableStateOf(RegistrationState())
+        private set
+
+    fun onEvent(event: RegistrationEvent) {
+        when (event) {
+            is RegistrationEvent.OnNameChange -> {
+                state = state.copy(
+                    name = event.name,
+                    nameError = false,
+                    validName = registrationUseCase.validFullname(event.name)
+                )
+            }
+
+            is RegistrationEvent.OnEmailChange -> {
+                state = state.copy(
+                    email = event.email,
+                    emailError = false,
+                    validEmail = registrationUseCase.validEmail(event.email)
```

Comment:
If you strictly want to stick to clean architecture, a use case should only have one public function. I'm personally not a fan of always doing that too strictly, but in this case I'd actually recommend against putting all registration validation functions in one class.
If you have a `RegistrationUseCase` that contains functions for validating email, password and full name, that means it's only used for registration (that's what the name says). Therefore, I'd either rename it to something like FormValidationManager or create specific classes for each validation type (like PasswordValidator or ValidatePasswordUseCase) - this will make them more reusable since you might need to validate an email/full name/password field outside of the registration screen.